### DISCUSSION TITLE
@tanstack/react-query prefetching implementation with V1

### DISCUSSION THREAD

**Discussion by JoshShearer** [vikejs/vike#1207] [Discussion]
Trying to implement prefetching for react-query for V1.  Encountering issues.

I have reviewed the following.
https://vike.dev/migration/v1-design#custom-hooks-exports
https://vike.dev/meta
https://vike.dev/header-file
https://github.com/vikejs/vike/discussions/774

Is there any way I can see a revision of this to V1?
https://github.com/YannBirba/vite-ssr-with-react-query

I am trying to allow access to /page/+onBeforeRender.ts
my current implementation of /page/+config.h.ts...
```
import type { Query, QueryKey } from "@tanstack/react-query";
import type { Config } from "vike/types";
import { api_firebase_speciesQuery } from "#src/api/firebase/speciesQuery";
import type { QueryPrefetch } from "#src/models/types";

const prefetchQueries: QueryPrefetch = [
    "species",
    {
      fn: (() => api_firebase_speciesQuery(3, [""], ["likes", "desc"])) as unknown as Query<unknown, Error, unknown, QueryKey>,
      options: {
        queryKey: ["species", "popularSpecies"],
      },
    },
  ];

const config: Config = {
  prefetchQueries,
  api_firebase_speciesQuery,
};


export default config;
```

**Comment by brillout** [comment]
Seems to be a duplicate of https://github.com/vikejs/vike/discussions/1207.
